### DISCUSSION TITLE
Feature request: Read TAP files

### DISCUSSION THREAD

**Discussion by Maverick-Shark** [frntc/Sidekick64#37] [Ideas]
Two feature request (related with TAP files):
1-Use Raspberry Pi as a 1541 unit (see Pi1541 project at https://cbm-pi1541.firebaseapp.com/)
2-Read .idx (index file) as a .tap file counter (datassatte counter) and jump to the correct possition

Hi,

I would like to load tap preserved files from cassettes with index.

In a .tap preserved file (tape), you have all the programs saved one per one (because all of them are recorded after each other in one preserved file).
Then, I would like to put the virtual position of each program (as the datassette counter) to run Turbo Load program at the beginning and jump to the correct position to load the program/game.

Another example, 007 A View To a Kill preserved tap game (it's only one file too). It has different missions and you need to reset your c64 after finish each one to load the following (because at the end there are completely independent programs) and when you are using a real datassette, you need to move to the correct position.

There is a tool named TAP Manager (v.1.6.0 or later), that you can download from http://www.manosoft.it/ that generates an index file (.idx). You can choose between to type of index files (Princess or Ultimate 64):

![image](https://user-images.githubusercontent.com/10196716/162678276-b9e9646a-b0a6-4411-9fb4-cb210dd8b557.png)

Thanks for your big job, its awesome!!
Shark

**Comment by frntc** [comment]
I have to admit that I never used .TAP-files. I'm not sure if I completely understand the use case: do you want a tape emulator that plays the signals stored in the .TAP-file (which would require some connection to the tape port)? Extracting PRGs from a .TAP might be possible, but I would have to find out if there's third-party libraries which implement turbo loaders. If the latter is what you're interested in, that might be easier (modulo turbo loaders)!

Regarding Pi1541: well this is tempting of course, but right now I don't see where/how to free up GPIOs for IEC-communication. Also, Pi1541 is implemented in another baremetal framework and porting is not trivial.

**Reply by Maverick-Shark** [reply]
Hi,
Thanks for your reply and for your spent time. And, yes, you're right, the solution is to add a tape emulator that sure it requires connection to the tape port (not a 1541 drive emulation). Could be possible?

Thanks and Best regards,
Shark

**Reply by frntc** [reply]
I'm not planning a tape emulator at the moment (isn't the idea of Sidekick64 to simplify/speed up some tasks? :-)). Maybe at some point in the future ;-)

(a true 1541 emulator would also require a connection to the IEC port, but as said, the (non-)available GPIOs are a hinderance)
